src/reversible_tokenize.jl

```diff
@@ -0,0 +1,54 @@
+# A simple reversible tokenizer
+
+MERGESYMBOL = '~'
+
+function is_weird(c::AbstractChar)
+    return .!(isletter(c) ||isnumeric(c) || isspace(c))
+end
+
+
+function rev_tokenize(instring::AbstractString)
+    ans = ""
+    for ind in eachindex(instring)
+        c = instring[ind]
+        c_p = ind > 1 ? instring[ind-1] : c
+        c_n = ind < length(instring) ? instring[ind+1] : c
+        
+        if .!(is_weird(c))
+            ans = ans*c
+        else
+            if .!(isspace(c_p))
+                ans = ans * " " * MERGESYMBOL
+            end
+            ans = ans * c
+            if .!(isspace(c_n) && is_weird(c_n))
+                ans = ans * MERGESYMBOL * " "
+            end
+        end
+    end
+    ans
+end
+
+
+function rev_detokenize(instring::AbstractString)
```

Comment:
I feel like this should take a array of strings

Reply:
@oxinabox Actually, the detokenization here is achieved by leaving markers in the original string and the thus the method takes input as a single string in a tokenized representation with markers.

Reply:
Yes, that is also a problem with the tokenzier.
A tokenizer must return an iterable collection of tokens.
That is how we define it.
Just adding `split` to your answer will do this though.